emData/generate_TC.py

```diff
@@ -0,0 +1,293 @@
+parametersFile.write("// each TrackletCalculator and that come directly from the wiring.\n")
+parametersFile.write("//\n")
+parametersFile.write("// The specific inner all-stub memory that the indices in a given stub-pair\n")
+parametersFile.write("// memory correspond to is determined by ASInnerMask. The bits of this mask,\n")
+parametersFile.write("// from least significant to most significant, represent the stub-pair memories\n")
+parametersFile.write("// in the order they are passed to TrackletCalculator; e.g., the LSB\n")
+parametersFile.write("// corresponds to stubPairs[0]. The bit for a given stub-pair memory is the\n")
+parametersFile.write("// index (0 or 1) of the inner all-stub memory that should be used. Likewise,\n")
+parametersFile.write("// the specific outer all-stub memories are determined by ASOuterMask.\n")
+parametersFile.write("//\n")
+parametersFile.write("// The validity of each of the barrel TPROJ memories is determined by\n")
+parametersFile.write("// TPROJMaskBarrel. The bits of this mask, from least significant to most\n")
+parametersFile.write("// significant, represent the memories in the order they are passed to\n")
+parametersFile.write("// TrackletCalculator; e.g., the LSB corresponds to\n")
+parametersFile.write("// projout_barrel_ps[TC::L1PHIA]. If a bit is set, the corresponding memory is\n")
+parametersFile.write("// valid, if it is not, the corresponding memory is not valid. Likewise, the\n")
+parametersFile.write("// validity of each of the disk TPROJ memories is determined by TPROJMaskDisk\n")
+parametersFile.write("// in the same way.\n")
+topHeaderFile = open("TrackletCalculatorTop.h", "w")
+topHeaderFile.write("#ifndef TrackletAlgorithm_TrackletCalculatorTop_h\n")
+topHeaderFile.write("#define TrackletAlgorithm_TrackletCalculatorTop_h\n")
+topHeaderFile.write("\n")
+topHeaderFile.write("#include \"TrackletCalculator.h\"\n")
+topFile = open("TrackletCalculatorTop.cc", "w")
+topFile.write("#include \"TrackletCalculatorTop.h\"\n")
+topFile.write("\n")
+topFile.write("////////////////////////////////////////////////////////////////////////////////\n")
+topFile.write("// Top functions for various TrackletCalculators (TC). For each iteration of\n")
+topFile.write("// the main processing loop, a TC retrieves a pair of stub indices from one of\n")
+topFile.write("// the stub-pair memories, and in turn, these indices are used to retrieve one\n")
+topFile.write("// stub each from an inner and an outer all-stub memory. This pair of stubs is\n")
+topFile.write("// used to calculate a rough set of helix parameters, which are written to the\n")
+topFile.write("// tracklet-parameter memory if the tracklet passes cuts on rinv and z0. Rough\n")
+topFile.write("// projections to additional layers and disks are also calculated and are\n")
+topFile.write("// written to the appropriate tracklet-projection memories.\n")
+topFile.write("////////////////////////////////////////////////////////////////////////////////\n")
```

Comment:
you might be able to simplify this piece of code by using triple quotes for a single, multi-line string, similar to the docstring. convention

Reply:
I have switched to multiline strings wherever possible. But I used parentheses instead of triple quotes so that I could indent the strings, which I think improves readability a little.